util/condvar/timedwait.go

```diff
@@ -42,7 +44,7 @@ func TimedWait(c *sync.Cond, timeout time.Duration) {
 			// thread hasn't gotten around to calling c.Wait()
 			// yet, so the c.Broadcast() did not wake it up.
 			// Sleep for a second and check again.
-			<-time.After(time.Second)
+			time.Sleep(time.Second)
```

Comment:
Why not use `util.NanoSleep()`?

Reply:
I used NanoSleep only in places that have time-criticality. On this one, we clearly don't care of it would take exactly 1 second or 1 second and 1 ms.